engine/pkg/rpcutil/client.go

```diff
@@ -98,11 +101,20 @@ func (c *FailoverRPCClients[T]) UpdateClients(ctx context.Context, urls []string
 
 		if _, ok := c.clients[addr]; !ok {
 			log.L().Info("add new server master client", zap.String("addr", addr))
-			cli, conn, err := c.dialer(ctx, addr)
+
+			var (
```

Comment:
Do we need to add `MaxTimeout`?

Reply:
The timeout is determined by implementation, ref：https://github.com/pingcap/tiflow/blob/master/engine/pkg/externalresource/manager/client.go#L29

Reply:
Have u test the timeout case? Do we have any error log if we have timeout here(or add some error log here), or we will face 15 seconds stuck here without acknowlege about what happens?

Reply:
Rest LGTM.